### ISSUE TITLE
SEXP warnings when combining "and" SEXP with "true" argument and setting event log flags

### ISSUE THREAD

**Issue by asarium** [scp-fs2open/fs2open.github.com#1176]
I know, the title is very confusing and I have no idea what those things have in common but I'll try to explain the bug as best as I can.

When debugging the capship command mission "The Great Escape" there are a lot of "ERROR: op_num function returned -1, this should not happen.." messages. I traced that to the "Warp done!" SEXP and then moved that to a separate mission to reproduce the bug. I can reliably reproduce the bug but as soon as I remove the line "+Event Log Flags: ( "true" )" the debug message goes away.

Here is the mission:  [SEXPBug.fs2](https://github.com/scp-fs2open/fs2open.github.com/files/750178/SEXPBug.fs2.txt)

Also, the bug has something to do with combining an "and" SEXP and a "true" constant. When I remove that and simply use the parameter of the SEXP that is relevant the message disappears again.

@Goober5000 Maybe you know what may be going on here? I think you were the one who implemented the SEXP log code.

**Comment by MageKing17** [comment]
I believe the log was coded by @Karajorma.

**Comment by asarium** [comment]
Ah, ok. I guess anyone who has experience with the SEXP code could be helpful here.

**Comment by asarium** [comment]
:man_facepalming: I just saw that there is a condition before the call that fails that checks if the event should be logged. I guess that explains why it doesn't happen if SEXP logging is disabled.

**Comment by asarium** [comment]
It looks like there is some kind of issue where the `text` of a SEXP node is not being set to the operator name which means that the lookup fails. The value of that SEXP node has the `SEXP_KNOWN_TRUE` value so maybe there is some kind of interaction with that.

**Comment by Goober5000** [comment]
Taking a look at this.

**Comment by Goober5000** [comment]
Ah, crap.  I *can't* take a look at this, due to #1051.  When the game hits the breakpoint, focus is returned to the debugger, but the window still shows FreeSpace.  I can interact with dialogs but I can't actually see what I'm doing.  I had to use the keyboard shortcut to exit the debugger in order to get back to normal.

**Comment by Goober5000** [comment]
With #1051 out of the way, I think I know what's going on.  The sexp node structure is a little weird in that, when dealing with a list, the CAR of the list refers to the operator and the CDR refers to the arguments.  This comes from Lisp, and CAR represents Sexp_nodes[n].first while CDR represents Sexp_nodes[n].rest.  I'll make a new PR.

**Comment by Goober5000** [comment]
I figured it out and have a solution ready to go.  Though I need to check with karajorma as to how to actually activate logging.  I PMmed him on HLP.

**Comment by asarium** [comment]
The test mission I uploaded uses event logging so you could use that to test your solution.

**Comment by Goober5000** [comment]
I did.  Although I see text written to the buffer when I trace through, nothing gets printed to the log file.

**Comment by MageKing17** [comment]
Are you looking at the right log file? The one in `%APPDATA%\HardLightProductions\FreeSpaceOpen\<mod>\data\`?

**Comment by Goober5000** [comment]
Well, not at first; I was looking under FreeSpace2\data.  But I just looked under the %APPDATA% folder and there wasn't much more:

```
02/11 23:48:03~   FS2_Open Mission Log - Opened 


Mission Untitled loaded.

```

Still none of the event logging I was expecting.

**Comment by Goober5000** [comment]
The event log might actually be broken currently.  I traced through deeply enough to find some mask values that weren't overlapping with each other.  The ~Q didn't work either.  I PMmed @Karajorma.

I'm pretty sure my solution should fix it, assuming I can get the event logging system in a state to test with.

**Comment by Goober5000** [comment]
PR opened: #1215

**Comment by Goober5000** [comment]
Okay, thanks to some troubleshooting with @karajorma, this looks like it's working now.  The event log wasn't refreshing while FSO was running, which confused things.

**Comment by asarium** [comment]
This was fixed by #1215.
